runtime/executor/src/test/java/edu/snu/nemo/runtime/executor/datatransfer/DataTransferTest.java

```diff
@@ -300,6 +303,15 @@ private void writeAndRead(final BlockManagerWorker sender,
     dummyIREdge.setProperty(DataPersistenceProperty.of(DataPersistenceProperty.Value.Keep));
     dummyIREdge.setProperty(EncoderProperty.of(ENCODER_FACTORY));
     dummyIREdge.setProperty(DecoderProperty.of(DECODER_FACTORY));
+    if (dummyIREdge.getPropertyValue(CommunicationPatternProperty.class).get()
```

Comment:
I'm not sure the `DataSkewMetricProperty` set in L306-L314, L398-L405 is used/checked afterwards. A standalone test may be a better option.

Reply:
It is actually used in `InputReader#read`(L355 and others). As now input data for each task is set as Execution Properties(`DefaultMetricPass`) instead of the constructor in `StageEdge`, we need to initialize it here for shuffle read.